### PR TITLE
Part of #19737: Replacing `FormField` with `FormTextField`

### PR DESCRIPTION
**Pull request by dhruvv173**
## Explanation

- This PR is a part of #19737 
- Made changes to the `AdvancedGasControl` component and added controls to the story for the same.
`ui/components/app/advanced-gas-controls`

console errors in the story:
image

### Before

image
image

### After

image
image

image

## Manual Testing Steps

1. Pull this branch
2. Run `yarn storybook`
3. Check `AdvancedGasControls` story and check console for errors

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by dhruvv173** [comment]
hey @georgewrmarshall, I think this PR a way to go before it can be considered for review. I would appreciate if you could help me out with some issues. Thank you!

File: ui/components/app/advanced-gas-controls/advanced-gas-controls.component.js [hunk 1]
```diff
@@ -29,23 +45,31 @@ export default function AdvancedGasControls({
...
         }}
-        tooltipText={t('editGasLimitTooltip')}
-        value={gasLimit}
-        allowDecimals={false}
-        numeric
       />
       <>
-        <FormField
-          titleText={t('advancedGasPriceTitle')}
-          titleUnit="(GWEI)"
```
Comment (dhruvv173) [inline ui/components/app/advanced-gas-controls/advanced-gas-controls.component.js:40]: how can we add `titleUnit` with label component?

File: ui/components/app/advanced-gas-controls/advanced-gas-controls.component.js [hunk 2]
```diff
@@ -61,7 +85,7 @@ AdvancedGasControls.propTypes = {
   onManualChange: PropTypes.func,
   gasLimit: PropTypes.number,
   setGasLimit: PropTypes.func,
-  gasPrice: PropTypes.string,
```
Comment (dhruvv173) [inline ui/components/app/advanced-gas-controls/advanced-gas-controls.component.js:64]: this should be `number` and not `string` right?

File: ui/components/app/advanced-gas-controls/advanced-gas-controls.component.js [hunk 3]
```diff
@@ -18,8 +20,22 @@ export default function AdvancedGasControls({
...
-        titleText={t('gasLimit')}
+      <Box
+        display={Display.Flex}
+        alignItems={AlignItems.center}
+        marginBottom={1}
+      >
+        <Label htmlFor="gas-limit" marginRight={1}>
+          Gas limit
+        </Label>
+        <InfoTooltip contentText={t('editGasLimitTooltip')} position="bottom" />
```
Comment (dhruvv173) [inline ui/components/app/advanced-gas-controls/advanced-gas-controls.component.js:31]: there have been visual changes to the title as we're using the `Label` component now, can you help fix this?

Comment (georgewrmarshall) [reply ui/components/app/advanced-gas-controls/advanced-gas-controls.component.js:31]: Hey @dhruvv173, there may be some slight visual changes that align with the newer components which is correct. It should be mostly font-size the layout should remain the same. I haven't checkout out this branch yet but if you are noticing some font-size/font-weight changes that should be ok. Will try to review early this week

File: ui/components/app/advanced-gas-controls/advanced-gas-controls.component.js (hunk 1 above)
Comment (georgewrmarshall) [reply ui/components/app/advanced-gas-controls/advanced-gas-controls.component.js:40]: Hey @dhruvv173, you may want to checkout this story and check the code for it https://metamask.github.io/metamask-storybook/?path=/story/components-componentlibrary-formtextfield--custom-label-or-help-text

**Comment by georgewrmarshall** [comment]
Converting to draft while we work on issues @dhruvv173 has outlined

File: ui/components/app/advanced-gas-controls/advanced-gas-controls.component.js (hunk 3 above)
Comment (dhruvv173) [reply ui/components/app/advanced-gas-controls/advanced-gas-controls.component.js:31]: Yes the concern was related to changes in font size/weight. Will wait for your review, thanks!

File: ui/components/app/advanced-gas-controls/advanced-gas-controls.component.js (hunk 1 above)
Comment (dhruvv173) [reply ui/components/app/advanced-gas-controls/advanced-gas-controls.component.js:40]: I referred to the same story and docs while working on this. But the current component displays unit as shown in the image below, still trying to figure out how we can add that, will look into it in the morning, thank you for your help!
image

File: ui/components/app/advanced-gas-controls/advanced-gas-controls.component.js (hunk 2 above)
Comment (georgewrmarshall) [reply ui/components/app/advanced-gas-controls/advanced-gas-controls.component.js:64]: Hmm not sure the component that is using it is also propTyped as a string so I think we can just leave it.

Comment (dhruvv173) [reply ui/components/app/advanced-gas-controls/advanced-gas-controls.component.js:64]: i changed it to a number, will revert the changes now, thanks for your input

File: ui/components/app/advanced-gas-controls/advanced-gas-controls.component.js (hunk 1 above)
Comment (dhruvv173) [reply ui/components/app/advanced-gas-controls/advanced-gas-controls.component.js:40]: completely forgot about the `Text` component which could be used to display the unit, thanks for the update!

File: ui/components/app/advanced-gas-controls/advanced-gas-controls.component.js (hunk 2 above)
Comment (dhruvv173) [reply ui/components/app/advanced-gas-controls/advanced-gas-controls.component.js]: Reverted changes here [`1ba4483`](https://github.com/MetaMask/metamask-extension/pull/19758/commits/1ba44835e6a04b9731f1b5304816f148bccf20b4)

**Comment by dhruvv173** [comment]
hey @georgewrmarshall, just wanted to ask if this approach of replacing the component is alright, if yes i will start with other components.

**Comment by georgewrmarshall** [comment]
Hey @dhruvv173, I believe so but as to get this PR across the line you will have to fix the broken e2e tests. I would suggest working on those before submitting another PR. Thanks 🙏

**Comment by dhruvv173** [comment]
> Hey @dhruvv173, I believe so but to get this PR merged you will have to fix the broken e2e tests. I would suggest working on those before submitting another PR. Thanks 🙏

still trying to find a way to fix the error related to lavamoat while building the test app. Will update if I find a way around or else we can close this one and I'll work on smaller issues from next time.

**Comment by dhruvv173** [comment]
> Hey @dhruvv173, I believe so but to get this PR merged you will have to fix the broken e2e tests. I would suggest working on those before submitting another PR. Thanks 🙏

From the [link](https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/49921/workflows/0ceabdd5-ba70-4e42-bf04-bf790f0b23d9/jobs/1435230?invite=true#step-106-471) for the failing e2e tests, shouldnt this css be `'.mm-box--flex-direction-row > h6'` ?

image

**Comment by georgewrmarshall** [comment]
> > Hey @dhruvv173, I believe so but to get this PR merged you will have to fix the broken e2e tests. I would suggest working on those before submitting another PR. Thanks 🙏
> 
> From the [link](https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/49921/workflows/0ceabdd5-ba70-4e42-bf04-bf790f0b23d9/jobs/1435230?invite=true#step-106-471) for the failing e2e tests, shouldnt this css be `'.mm-box--flex-direction-row > h6'` ?
> 
> image

Yes, I believe so!

**Comment by dhruvv173** [comment]
> > > Hey @dhruvv173, I believe so but to get this PR merged you will have to fix the broken e2e tests. I would suggest working on those before submitting another PR. Thanks 🙏
> > 
> > 
> > From the [link](https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/49921/workflows/0ceabdd5-ba70-4e42-bf04-bf790f0b23d9/jobs/1435230?invite=true#step-106-471) for the failing e2e tests, shouldnt this css be `'.mm-box--flex-direction-row > h6'` ?
> > image
> 
> Yes, I believe so!

tried it [`here`](https://github.com/MetaMask/metamask-extension/pull/19758/commits/46caea86cd40f8f807860702f388f48dab0840d1) but it didnt work so reverted it :/
It would be great if someone can assist with e2e tests or else we can close this.
Thanks!

**Comment by georgewrmarshall** [comment]
Hey @dhruvv173, could you try rebasing this PR and see if that helps with the e2e tests?

**Comment by dhruvv173** [comment]
> Hey @dhruvv173, could you try rebasing this PR and see if that helps with the e2e tests?

done, still the same :/

**Comment by georgewrmarshall** [comment]
Hey @dhruvv173, I really appreciate this contribution but after further review it looks like the `NumericInput` is handling some logic that isn't present in the `TextField` components so I think we need to replace some of the components there first

**Pull request closed without merging**